src/librepgp/stream-armor.c

```diff
@@ -952,6 +901,22 @@ init_armored_dst(pgp_dest_t *dst, pgp_dest_t *writedst, pgp_armored_msg_t msgtyp
     return ret;
 }
 
+bool
+is_armored_source(pgp_source_t *src)
+{
+    const char armor_start[] = "-----BEGIN PGP";
+    uint8_t    buf[128];
+    ssize_t    read;
+
+    read = src_peek(src, buf, sizeof(buf));
+    if (read < sizeof(armor_start)) {
```

Comment:
Thanks for spotting this. Actually, problem can be wider since I used signed vs unsigned comparison in many places believing that signed will not be converted. Will re-check code.

Reply:
...looks interesting: For C++, -Wsign-compare is included in the -Wall list of warnings, but not for C. For both gcc and clang -Wsign-compare is included in the -Wextra list of warnings.